sdv/constraints/tabular.py

```diff
@@ -388,3 +390,155 @@ def reverse_transform(self, table_data):
+        If it is a float, returns the value itself.
+
+        Args:
+            table_data (pandas.DataFrame):
+                The Table data.
+
+        Returns:
+            pandas.DataFrame or float:
+                The upper bound.
+        """
+        if isinstance(self._high, str):
+            return table_data[self._high]
+
+        return self._high
+
+    def is_valid(self, table_data):
+        """Say whether the ``constraint_column`` is between the ``low`` and ``high`` values.
+
+        Args:
+            table_data (pandas.DataFrame):
+                Table data.
+
+        Returns:
+            pandas.Series:
+                Whether each row is valid.
+        """
+        if self._strict:
+            satisfy_low_bound = self._get_low(table_data) < table_data[self.constraint_column]
+            satisfy_high_bound = table_data[self.constraint_column] < self._get_high(table_data)
+            return satisfy_low_bound & satisfy_high_bound
+
+        satisfy_low_bound = self._get_low(table_data) <= table_data[self.constraint_column]
+        satisfy_high_bound = table_data[self.constraint_column] <= self._get_high(table_data)
+        return satisfy_low_bound & satisfy_high_bound
+
+    def transform(self, table_data):
+        """Transform the table data.
+
+        The transformation consists of scaling the ``constraint_column``
+        (``(column-low)/(high-low) * cnt + small_cnt``) and then applying
+        a logit function to the scaled version of the column.
+
+        Args:
+            table_data (pandas.DataFrame):
+                Table data.
+
+        Returns:
+            pandas.DataFrame:
+                Transformed data.
+        """
+        table_data = table_data.copy()
+        low = self._get_low(table_data)
+        high = self._get_high(table_data)
+
+        data = (table_data[self.constraint_column] - low) / (high - low)
+        data = data * 0.95 + 0.025
+        data = np.log(data / (1.0 - data))
+
+        table_data[self._transformed_column] = data
+        table_data = table_data.drop(self.constraint_column, axis=1)
+
+        return table_data
+
+    def reverse_transform(self, table_data):
+        """Reverse transform the table data.
+
+        The reverse transform consists of applying a sigmoid to the transformed
+        ``constraint_column`` and then scaling it back to the original space
+        ( ``(column - cnt) * (high - low) / cnt + low`` ).
+
+        Args:
+            table_data (pandas.DataFrame):
+                Table data.
+
+        Returns:
+            pandas.DataFrame:
+                Transformed data.
+        """
+        table_data = table_data.copy()
+        low = self._get_low(table_data)
+        high = self._get_high(table_data)
+        data = table_data[self._transformed_column]
+
+        data = 1 / (1 + np.exp(-data))
+        data = (data - 0.025) / 0.95
+        data = data * (high - low) + low
+        data = data.clip(low, high)
```

Comment:
We talked about this with @csala but I'm curious how much this transformation reduces the need to clip from say just modeling the column without a transformation and clipping the sampled value to the bounds directly

Reply:
It really depends on the constants you use when scaling: if you increase the multiplier from 0.95 to 0.99, it removes virtually all out of bounds values, but then the model just doesn't produce any values close to the tails. If you move the constants the other way, the opposite happens. I didn't play around with it too much so there may be a better set of constants that balance the number of out of bound values/performance ratio.